src/rust/engine/src/nodes/mod.rs

```diff
@@ -369,12 +369,31 @@ impl NodeKey {
     }
 
     async fn maybe_watch(&self, context: &Context) -> NodeResult<()> {
-        if let Some((path, watcher)) = self.fs_path_to_watch().zip(context.core.watcher.as_ref()) {
-            let abs_path = context.core.build_root.join(path);
-            watcher
-                .watch(abs_path)
-                .map_err(|e| Context::mk_error(&e))
-                .await
+        if let Some(path) = self.fs_path_to_watch() {
```

Comment:
`fs_path_to_watch` should probably return an enum so it becomes easier to distinguish buildroot-confined paths from system paths (and the rule API would need to accommodate that change as well).